feathr_project/feathr/registry/feature_registry.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC, abstractmethod
+
+from typing import Any, Dict, List, Optional, Tuple
+from feathr.definition.feature_derivations import DerivedFeature
+from feathr.definition.anchor import FeatureAnchor
+
+class FeathrRegistry(ABC):
+    """This is the abstract class for all the feature registries. All the feature registries should implement those interfaces.
+    """
+
+    @abstractmethod
+    def register_features(self, from_context: bool = True):
+        """Registers features based on the current workspace
+
+                Args:
+                from_context: If from_context is True (default), the features will be generated from
+                    the current context, with the previous built features in client.build(). Otherwise, the features will be generated from
+                    configuration files.
+        """
+        pass
+
+
+    @abstractmethod
+    def list_registered_features(self, project_name: str = None) -> List[str]:
```

Comment:
Just listing the feature names might not be that useful. Can we remove this API or improve it?

Reply:
let's keep it there since it's an easy way to verify all the features are registered. We should encourage end users to use `get_features_from_registry` more.